projects/ngx-location-picker/src/lib/services/ngx-location-picker.service.ts

```diff
@@ -22,130 +24,132 @@ import {CoordinateModel} from '../types/coordinate.model';
-
-        return this.httpClient.get<LocationModel[]>(`${this.locationPickerApi}/locations`, {params: parameters});
-    }
-
-    /**
-     * Search addresses based on street name and house number
-     *
-     * @param query (the address to look for, consists of a street name and house number)
-     *
-     * @return Observable<AddressModel[]>
-     */
-    private searchAddresses(query: AddressQueryModel): Observable<AddressModel[]> {
-        const parameters = this.locationPickerHelper.toHttpParams(query);
-
-        return this.httpClient.get<AddressModel[]>(`${this.locationPickerApi}/addresses`, {params: parameters});
-    }
-
-    /**
-     * Get an address by id
-     *
-     * @param query (the address id to get)
-     *
-     * @return Observable<AddressModel[]>
-     */
-    private searchAddressById(query: AddressIdQueryModel): Observable<AddressModel[]> {
-        const parameters = this.locationPickerHelper.toHttpParams(query);
-
-        return this.httpClient.get<AddressModel[]>(`${this.locationPickerApi}/addresses/${query.id}`, {params: parameters});
-    }
-
-    /**
-     * Search locations based on a set of coordinates
-     *
-     * @param query (search query containing coordinates and other parameters)
-     *
-     * @return Observable<CoordinateModel[]>
-     */
-    private searchLocationsByCoordinates(query: CoordinateQueryModel): Observable<CoordinateModel[]> {
-        const parameters = this.locationPickerHelper.toHttpParams(query);
-
-        return this.httpClient.get<any>(`${this.locationPickerApi}/coordinates`, {params: parameters});
+  private locationPickerApi = '';
+
+  /**
+   * NgxLocationPickerService constructor
+   *
+   * @param httpClient (httpClient used for making requests to the api)
+   * @param locationPickerHelper (Helper functions class)
+   */
+  constructor(
+    private httpClient: HttpClient,
+    private locationPickerHelper: NgxLocationPickerHelper
+  ) {
+  }
+
+  /**
+   * Main search function. Determines which action to undertake based on the provided search string.
+   *
+   * @param search (the search query)
+   * @param baseUrl (required the url to the BFF)
+   * @param limit (the amount of locations to return, only used for querying searchLocations)
+   * @param layers (the layers to look for locations in, only used for querying searchLocations)
+   * @param prioritizelayer (the layer to boost)
+   * @param sort (key to sort results by)
+   * @param cascadingReturnSingle (whether or not to return a single cascading result)
+   * @param cascadingRules (the configuration for doing reverse lookups by coordinates)
+   *
+   * @return Observable<LocationModel[] | AddressModel[] | CoordinateModel[]>
+   */
+  delegateSearch(
+    search: DelegateSearchModel
+  ): Observable<LocationModel[] | AddressModel[] | CoordinateModel[]> {
+    search.prioritizelayer = search.prioritizelayer ? search.prioritizelayer : ['straatnaam'];
```

Comment:
Due to the refactor of the params to searchmodel added null check on prioritizelayer to determine if it has a value otherwise the default value is filled in.

I've noticed I still have to change the params documentation